### PR TITLE
Step 2: Added unit tests for in-memory services and controllers

### PR DESCRIPTION
**Pull request by Tonya-Lyub**
This pull request implements Step 2 of the project requirements:

Added unit tests for TaskService, UserService, and NotificationService.

Added tests for TaskController, UserController, and NotificationController endpoints.

Used JUnit 5 and Mockito for mocking dependencies and verifying behavior.

Ensured coverage for main CRUD operations and edge cases (e.g., marking tasks as deleted, user login).

Tests use the existing in-memory data storage implementation (no database yet).

Tests are organized in separate test classes following best practices.

### CODE REVIEW

File: src/test/java/com/example/taskmanager/service/UserServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+        // Assert
+        assertTrue(result.isPresent());
+        assertEquals("testuser", result.get().getUsername());
+    }
+
+    @Test
+    void findByUsername_ShouldReturnEmpty_WhenUserDoesNotExist() {
+        // Arrange
+        when(userStorage.findByUsername("nonexistent")).thenReturn(Optional.empty());
```
Comment (Nenad7955) [inline src/test/java/com/example/taskmanager/service/UserServiceTest.java:72]: этот тест ничего не проверяет..
